near/src/config.rs

```diff
@@ -604,26 +607,56 @@ impl GenesisConfig {
     }
 
     /// Reads GenesisConfig from a file.
-    pub fn from_file(path: &PathBuf) -> Self {
+    pub fn from_file(path: &PathBuf, records_path: Option<PathBuf>) -> Self {
         let mut file = File::open(path).expect("Could not open genesis config file.");
         let mut content = String::new();
         file.read_to_string(&mut content).expect("Could not read from genesis config file.");
-        GenesisConfig::from(content.as_str())
+        let records_str = records_path.map(|path| {
+            let mut content = String::new();
+            let mut file = File::open(&path).expect("Could not open genesis records file.");
+            file.read_to_string(&mut content).expect("Could not read from genesis records file.");
+            content
+        });
+        GenesisConfig::from(content.as_str(), records_str)
     }
 
     /// Writes GenesisConfig to the file.
     pub fn write_to_file(&self, path: &Path) {
         let mut file = File::create(path).expect("Failed to create / write a genesis config file.");
-        let str =
-            serde_json::to_string_pretty(self).expect("Error serializing the genesis config.");
-        if let Err(err) = file.write_all(str.as_bytes()) {
+        let mut buf =
+            serde_json::to_vec_pretty(self).expect("Error serializing the genesis config.");
+        if let Err(err) = file.write_all(&mut buf) {
             panic!("Failed to write a genesis config file {}", err);
         }
     }
-}
 
-impl From<&str> for GenesisConfig {
-    fn from(config: &str) -> Self {
+    /// Writes GenesisConfig to the 2 files.
+    /// All fields but records into `config_path` file and records into a `records_path` file.
+    /// Requires `&mut self` to avoid cloning records.
+    pub fn write_to_config_and_records_files(&mut self, config_path: &Path, records_path: &Path) {
```

Comment:
I don't think that's intuitive when writing config to files modifies the config (even temporarily). To avoid this we need to have a separation similar to `NearConfig`/`Config`. Where one struct is used to be serialized and deserialized from file (`Config`) and another struct is constructed from it (`NearConfig`).

Reply:
Since modifying it now would break legacy configs, I preferred this solution as the minimum viable option.
2 alternatives that don't require `&mut` and don't modify the config structure are: cloning entire config or cloning all fields except for records.